src/components/ciPipeline/CiPipelineSourceConfig.tsx

```diff
@@ -1,89 +1,141 @@
-import React, { useState, useEffect } from 'react';
-import branchIcon from '../../assets/icons/misc/branch.svg';
-import webhookIcon from '../../assets/icons/misc/webhook.svg';
-import Tippy from '@tippyjs/react';
-import { SourceTypeMap } from '../../config';
-import { getWebhookEventsForEventId } from '../../services/service';
-import { ReactComponent as Info } from '../../assets/icons/ic-info-outlined.svg';
+import React, { useState, useEffect } from 'react'
+import branchIcon from '../../assets/icons/misc/branch.svg'
+import webhookIcon from '../../assets/icons/misc/webhook.svg'
+import Tippy from '@tippyjs/react'
+import { SourceTypeMap } from '../../config'
+import { getWebhookEventsForEventId } from '../../services/service'
+import { ReactComponent as Info } from '../../assets/icons/ic-info-outlined.svg'
+import regexIcon from '../../assets/icons/misc/regex.svg'
+export interface CIPipelineSourceConfigInterface {
+    sourceType
+    sourceValue
+    showTooltip?: boolean
+    showIcons?: boolean
+    baseText?: string
+    regex?: any
+    isRegex?: boolean
+    primaryBranchAfterRegex?: string
+}
 
-export function CiPipelineSourceConfig({ sourceType, sourceValue, showTooltip, showIcons = true, baseText = undefined }) {
+export function CiPipelineSourceConfig({
+    sourceType,
+    sourceValue,
+    showTooltip,
+    showIcons = true,
+    baseText = undefined,
+    regex,
+    isRegex,
+    primaryBranchAfterRegex,
+}: CIPipelineSourceConfigInterface) {
+    let _isWebhook = sourceType === SourceTypeMap.WEBHOOK
+    let _isRegex = sourceType === SourceTypeMap.BranchRegex || !!regex || isRegex
```

Comment:
Can we rely only on sourceType === SourceTypeMap.BranchRegex?

Reply:
No, we can't. Once will add a fixed value for regex then the source type will change to fixed